data/transactions/logic/resources.go

```diff
@@ -0,0 +1,356 @@
+// Copyright (C) 2019-2023 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package logic
+
+import (
+	"fmt"
+
+	"github.com/algorand/go-algorand/data/basics"
+	"github.com/algorand/go-algorand/data/transactions"
+	"github.com/algorand/go-algorand/ledger/ledgercore"
+	"github.com/algorand/go-algorand/protocol"
+)
+
+// resources contains a catalog of available resources. It's used to track the
+// apps, assets, and boxes that are available to a transaction, outside the
+// direct foreign array mechanism.
+type resources struct {
+	// These resources were created previously in the group, so they can be used
+	// by later transactions.
+	createdAsas []basics.AssetIndex
+	createdApps []basics.AppIndex
+
+	// These resources have been mentioned by some txn in the group, so they are
```

Comment:
a resource is "mentioned" when it is in the direct foreign array of any transaction in a group -- or are there other ways

Reply:
I'll change the wording to be more consistent, rather than say "mentioned".  Other txn types also make resources available, for example an axfer makes the accounts and asset used available.